beacon-chain/db/kv/state.go

```diff
@@ -196,6 +204,18 @@ func (k *Store) DeleteState(ctx context.Context, blockRoot [32]byte) error {
 			}
 		}
 
+		enc = bkt.Get(blockRoot[:])
+		if enc == nil {
+			return nil
+		}
+		block := &ethpb.SignedBeaconBlock{}
+		if err := decode(enc, block); err != nil {
+			return err
+		}
+		if err := k.clearStateSlotBitField(ctx, tx, block.Block.Slot); err != nil {
```

Comment:
Panic here if block is nil for whatever reason

Reply:
Handled in `slotByBlockRoot` above